### PR TITLE
SealProofVariant and lotus-bench NI-PoRep additions

### PR DESCRIPTION
**Pull request by rvagg**
* Introduce a `SealProofVariant` type with standard, synthetic and non-interactive forms so we can do simpler switching between the variants
* Wire up ni-porep to lotus-bench `simple` and add a configurable bash script that can run through the whole sealing process

### CODE REVIEW

File: cmd/lotus-bench/bench-sectors.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
```
Comment (BigLep) [inline cmd/lotus-bench/bench-sectors.sh:1]: I don't know the team history or view or all the context here, but in general would vote for not seeing this amount of bash getting checked in.

I generally advocate for a "two finger rule" with checked-in bash (i.e., that any bash that extends what I can cover with two fingers on the screen is a signal that instead convert to something else).  This is stemming from a code review and maintenance regard given (my opinion) shell scripting's esoteric and less-ergonomic nature.  

Basically if this is something we care about continuing to work and be modifiable in future, then I would convert it to something else (presumably go) that the team is well-versed in.

Comment (rvagg) [reply cmd/lotus-bench/bench-sectors.sh:1]: This is here for the reverse reason! I find this more ergonomic for doing a full seal workflow than what's available in lotus-bench today. I suppose it could be replaced with a `full` but I'm not sure who wants this. I mostly checked this in because I found it useful and didn't want to either discard it or put it in a place where it could get lost; but perhaps it's not useful or interesting to anyone else. I don't know!

Comment (BigLep) [reply cmd/lotus-bench/bench-sectors.sh:1]: Got it.  I like the idea of checking it in so we don't lose it.  It seems like we don't know the full utility of this in the long term so can leave it as is.  If this is something we need to use/maintain/support in future, we could make it more robust.  Seems good to me.

Comment (rvagg) [reply cmd/lotus-bench/bench-sectors.sh:1]: I'll add some additional notes to this—it may turn out to not be useful and if it's not touched in 6 months then maybe it gets thrown out, that's OK.

Comment (rvagg) [reply cmd/lotus-bench/bench-sectors.sh:1]: I've added some docs to the top of the script. We'll see what @aarshkshah1992 thinks of it being in here. I'm fine removing it entirely if you both think it's cruft.

File: itests/kit/ensemble.go [hunk 2]
```diff
@@ -260,7 +260,7 @@ func (n *Ensemble) MinerEnroll(minerNode *TestMiner, full *TestFullNode, opts ..
 		)
 
 		// Will use 2KiB sectors by default (default value of sectorSize).
-		proofType, err := miner.SealProofTypeFromSectorSize(options.sectorSize, n.genesis.version, false)
+		proofType, err := miner.SealProofTypeFromSectorSize(options.sectorSize, n.genesis.version, miner.SealProofVariant_Standard)
```
Comment (aarshkshah1992) [inline itests/kit/ensemble.go:263]: Why are we always using `Standard` in the ensemble ? What if tests wanna use NI ?

File: cmd/lotus-bench/bench-sectors.sh (hunk 1 above)
Comment (aarshkshah1992) [reply cmd/lotus-bench/bench-sectors.sh:1]: I had a look at this. Okay to keep it here if it's just to help out people who might want to use this as a reference to understand the full PoRep/NI-PoRep workflow. The script is straight forward to follow.

However, strongly recommend that we translate this to Golang if we're actually using this for benchmarking that workflow.

File: itests/kit/ensemble.go (hunk 2 above)
Comment (rvagg) [reply itests/kit/ensemble.go:263]: then someone's going to have to add it - this is for the full lotus-miner and it has no support for dealing with NI, it assumes precommit for a start, so currently it doesn't make sense to parameterise this

**Review by aarshkshah1992 — APPROVED** [review]
